### ISSUE TITLE
Upgrade to Eigen 3.4

### ISSUE THREAD

**Issue by ChipKerchner** [stan-dev/stan#3145]
#### Summary:
Quite a bit of improvements have been made to Eigen since 3.3.9.  In particular, Altivec (IBM Power) platform has made significant performance enhancement since.  Please consider upgrading Eigen to the latest version.

#### Description:
Replace Eigen libraries 3.3.9 with 3.4.x

#### Current Version:
v2.30.0

**Comment by RajalakshmiSR** [comment]
@WardBrian Do we have any target date for this 3.4 upgrade? Thanks.

**Comment by WardBrian** [comment]
@RajalakshmiSR No, we do not have a target date. It will be upgraded as soon as we are able to integrate it without existing behavior changing. At the moment this means it will not be upgraded until one of the developers is able to volunteer the time needed to find the remaining issues.
